core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BoundedRangeFileInputStream.java

```diff
@@ -62,12 +62,15 @@ public BoundedRangeFileInputStream(FSDataInputStream in, long offset, long lengt
 
   @Override
   public int available() throws IOException {
-    int avail = in.available();
-    if (pos + avail > end) {
-      avail = (int) (end - pos);
-    }
+    final FSDataInputStream inLocal = in;
```

Comment:
Very likely.  I was following the pattern down in the read method.  However looking more closely, the final mechanism was only used because of the wrapped run method.  I will remove the extra assignment.